packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx

```diff
@@ -1,49 +1,118 @@
-import React from 'react'
+import React, {useMemo, useState} from 'react'
 import graphql from 'babel-plugin-relay/macro'
+import useMenu from '../../../../hooks/useMenu'
 import styled from '@emotion/styled'
 import Row from '../../../../components/Row/Row'
 import Panel from '../../../../components/Panel/Panel'
 import {ElementWidth} from '../../../../types/constEnums'
-import {useFragment} from 'react-relay'
+import {PreloadedQuery, usePreloadedQuery} from 'react-relay'
 import OrgTeamsRow from './OrgTeamsRow'
-import {OrgTeams_organization$key} from '../../../../__generated__/OrgTeams_organization.graphql'
+import Menu from '../../../../components/Menu'
+import {MenuPosition} from '../../../../hooks/useCoords'
+import MenuItem from '../../../../components/MenuItem'
+import DashFilterToggle from '../../../../components/DashFilterToggle/DashFilterToggle'
+import {OrgTeamsQuery} from '../../../../__generated__/OrgTeamsQuery.graphql'
 
 const StyledPanel = styled(Panel)({
   maxWidth: ElementWidth.PANEL_WIDTH
 })
 
 type Props = {
-  organizationRef: OrgTeams_organization$key
+  queryRef: PreloadedQuery<OrgTeamsQuery>
 }
 
 const OrgTeams = (props: Props) => {
-  const {organizationRef} = props
-  const organization = useFragment(
+  const {queryRef} = props
+  const data = usePreloadedQuery<OrgTeamsQuery>(
     graphql`
-      fragment OrgTeams_organization on Organization {
-        id
-        isBillingLeader
-        teams {
-          id
-          ...OrgTeamsRow_team
+      query OrgTeamsQuery($orgId: ID!) {
+        viewer {
+          featureFlags {
+            canViewTeamsInDomain
+          }
+          organization(orgId: $orgId) {
+            id
+            activeDomain
+            isBillingLeader
+            name
+            teams {
+              id
+              ...OrgTeamsRow_team
+            }
+          }
+          domains {
+            id
+            organizations {
+              teams {
+                id
+                ...OrgTeamsRow_team
+              }
+            }
+          }
         }
       }
     `,
-    organizationRef
+    queryRef
   )
-  const {teams, isBillingLeader} = organization
+
+  const {togglePortal, originRef, menuPortal, menuProps} = useMenu(MenuPosition.UPPER_RIGHT)
+  const {viewer} = data
+  const {featureFlags, domains, organization} = viewer
+  const {activeDomain, teams, isBillingLeader, name} = organization ?? {}
+
+  const ALL_TEAMS_IN_ORG = `All Teams In ${name}`
```

Comment:
Originally, I added these labels to `FilterLabels` in `constEnums`, but I think we should show the org/domain name, so I'm instantiating them here